addons/format_lame.c

```diff
@@ -0,0 +1,559 @@
+		if (ret < 0) {
+			ast_log(LOG_ERROR, "LAME encoder initialization failed: %d", ret);
+			return -1;
+		}
+	}
+
+	res = lame_encode_buffer(p->lgfp, f->data.ptr, f->data.ptr, f->samples, encoder_buffer, sizeof(encoder_buffer));
+	if (res < 0) {
+		ast_log(LOG_WARNING, "LAME encoder returned error: %d", res);
+		return -1;
+	} else if (res > 0) {
+		wres = fwrite(encoder_buffer, 1, res, fs->f);
+		if (wres != res) {
+			ast_log(LOG_WARNING, "Bad write (%d/%d): %s", wres, res, strerror(errno));
+			return -1;
+		} else {
+			ast_debug(3, "LAME encoder wrote %d bytes", wres);
+		}
+	}
+
+	return 0;
+}
+
+static off_t mp3lame_tell(struct ast_filestream *s)
+{
+	struct mp3lame_private *p = s->_private;
+
+	if (p->decoder_counter != 0) {
+		return p->doffset;
+	} else {
+		return ftello(s->f) / BITS;
+	}
+}
+
+static int mp3lame_seek(struct ast_filestream *s, off_t sample_offset, int whence)
+{
+	struct mp3lame_private *p = s->_private;
+
+	off_t offset = 0;
+	off_t min = 0;
+	off_t max;
+	off_t current;
+
+	sample_offset <<= 1;
+
+	if (p->decoder_counter != 0) {
+		max = p->total_bytes_decoded;
+		current = p->doffset;
+	} else {
+		if ((current = ftello(s->f)) < 0) {
+			ast_log(AST_LOG_WARNING, "Unable to determine current position in mp3 filestream %p: %s", s, strerror(errno));
+			return -1;
+		}
+
+		if (fseeko(s->f, 0, SEEK_END) < 0) {
+			ast_log(AST_LOG_WARNING, "Unable to seek to end of mp3 filestream %p: %s", s, strerror(errno));
+			return -1;
+		}
+
+		if ((max = ftello(s->f)) < 0) {
+			ast_log(AST_LOG_WARNING, "Unable to determine max position in mp3 filestream %p: %s", s, strerror(errno));
+			return -1;
+		}
+	}
+
+	if (whence == SEEK_SET)
+		offset = sample_offset;
+	else if (whence == SEEK_CUR || whence == SEEK_FORCECUR)
+		offset = sample_offset + current;
+	else if (whence == SEEK_END)
+		offset = max - sample_offset;
+	if (whence != SEEK_FORCECUR) {
+		offset = (offset > max)?max:offset;
+	}
+	/* always protect against seeking past begining. */
+	offset = (offset < min)?min:offset;
+
+	if (p->decoder_counter != 0) {
+		p->doffset = offset;
+		p->dsamples = (p->total_bytes_decoded - offset) / BITS;
+
+		return 0;
+	}
+
+	return fseeko(s->f, offset, SEEK_SET);
+}
+
+static int mp3lame_trunc(struct ast_filestream *s)
+{
+	int fd;
+	struct mp3lame_private *p = s->_private;
+	off_t current = mp3lame_tell(s);
+
+	if (p->decoder_counter != 0) {
+		p->dsamples = 0;
+		p->total_bytes_decoded = p->doffset;
+		return 0;
+	} else {
+		if ((fd = fileno(s->f)) < 0) {
+			ast_log(AST_LOG_WARNING, "Unable to determine file descriptor for mp3 filestream %p: %s", s, strerror(errno));
+			return -1;
+		}
+		if ((current = ftello(s->f)) < 0) {
+			ast_log(AST_LOG_WARNING, "Unable to determine current position in mp3 filestream %p: %s", s, strerror(errno));
+			return -1;
+		}
+		return ftruncate(fd, current);
+	}
+}
+
+static char *mp3lame_getcomment(struct ast_filestream *s)
+{
+	char * comment = "Asterisk MP3 lame\0";
```

Comment:
No space between * and comment
I don't think the `\0` in the string is needed either